elytron/src/main/java/org/wildfly/extension/elytron/AuditResourceDefinitions.java

```diff
@@ -207,8 +219,65 @@ static ResourceDefinition getFileAuditLogResourceDefinition() {
     }
 
-    static ResourceDefinition getRotatingFileAuditLogResourceDefinition() {
-        AttributeDefinition[] attributes = new AttributeDefinition[] {PATH, RELATIVE_TO, SYNCHRONIZED, FORMAT, MAX_BACKUP_INDEX, ROTATE_ON_BOOT, ROTATE_SIZE, SUFFIX };
+    static ResourceDefinition getPeriodicRotatingFileAuditLogResourceDefinition() {
+        AttributeDefinition[] attributes = new AttributeDefinition[] {PATH, RELATIVE_TO, SYNCHRONIZED, FORMAT, PERIODIC_SUFFIX };
+        AbstractAddStepHandler add = new TrivialAddHandler<SecurityEventListener>(SecurityEventListener.class, attributes, SECURITY_EVENT_LISTENER_RUNTIME_CAPABILITY) {
+
+            @Override
+            protected ValueSupplier<SecurityEventListener> getValueSupplier(
+                    ServiceBuilder<SecurityEventListener> serviceBuilder, OperationContext context, ModelNode model)
+                    throws OperationFailedException {
+
+                final boolean synv = SYNCHRONIZED.resolveModelAttribute(context, model).asBoolean();
+                final Format format = Format.valueOf(FORMAT.resolveModelAttribute(context, model).asString());
+                final String suffix = PERIODIC_SUFFIX.resolveModelAttribute(context, model).asString();
+
+                final InjectedValue<PathManager> pathManager = new InjectedValue<>();
+
+                final String path = PATH.resolveModelAttribute(context, model).asString();
+                final String relativeTo = asStringIfDefined(context, RELATIVE_TO, model);
+
+                if (relativeTo != null) {
+                    serviceBuilder.addDependency(PathManagerService.SERVICE_NAME, PathManager.class, pathManager);
+                    serviceBuilder.addDependency(pathName(relativeTo));
+                }
+
+                return () -> {
+                    PathResolver pathResolver = pathResolver();
+                    pathResolver.path(path);
+                    if (relativeTo != null) {
+                        pathResolver.relativeTo(relativeTo, pathManager.getValue());
+                    }
+                    File resolvedPath = pathResolver.resolve();
+
+                    final SecurityEventVisitor<?, String> formatter = Format.JSON == format ? JsonSecurityEventFormatter.builder().setDateFormatSupplier(bind(SimpleDateFormat::new, DATE_FORMAT)).build() : SimpleSecurityEventFormatter.builder().setDateFormatSupplier(bind(SimpleDateFormat::new, DATE_FORMAT)).build();
+                    AuditEndpoint endpoint;
+                    try {
+                        FileAuditEndpoint.Builder builder =  PeriodicRotatingFileAuditEndpoint.builder()
+                                .setSuffix(suffix)
+                                .setLocation(resolvedPath.toPath())
+                                .setSyncOnAccept(synv)
+                                .setDateFormatSupplier(bind(SimpleDateFormat::new, DATE_FORMAT));
```

Comment:
See my comment on https://github.com/wildfly-security/wildfly-elytron/pull/845#issuecomment-306260856. We may want this to use a `java.time.format.DateTimeFormatter`. We're using JSR 310 to format the dates so we should probably use the formatter from the `java.time` API.